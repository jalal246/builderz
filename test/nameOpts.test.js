import { resolve } from "path";

import perFile from "./utils";

describe.only.each`
  pkgName                 | desc
  ${"name-empty"}         | ${"package.json without name"}
  ${"name-camelized"}     | ${"output name is camelized"}
  ${"name-not-camelized"} | ${"output isn't camelized"}
`("tests $pkgName for testing: $desc", ({ pkgName }) => {
  it("passes", async () => {
    const pathPure = resolve(__dirname, "fixtures", pkgName);
    const distPath = resolve(pathPure, "dist");

    await perFile(pathPure, distPath);
  });
});
