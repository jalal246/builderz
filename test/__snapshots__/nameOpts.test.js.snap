// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests name-camelized for testing: output name is camelized passes 1`] = `1`;

exports[`tests name-camelized for testing: output name is camelized passes 2`] = `
"var foo = function foo() {
  for (var _len = arguments.length, bar = new Array(_len), _key = 0; _key < _len; _key++) {
    bar[_key] = arguments[_key];
  }

  return Promise.resolve(\\"foo \\".concat(bar));
};

module.exports = foo;
"
`;

exports[`tests name-empty for testing: package.json without name passes 1`] = `1`;

exports[`tests name-empty for testing: package.json without name passes 2`] = `
"var foo = function foo() {
  for (var _len = arguments.length, bar = new Array(_len), _key = 0; _key < _len; _key++) {
    bar[_key] = arguments[_key];
  }

  return Promise.resolve(\\"foo \\".concat(bar));
};

module.exports = foo;
"
`;

exports[`tests name-not-camelized for testing: output isn't camelized passes 1`] = `1`;

exports[`tests name-not-camelized for testing: output isn't camelized passes 2`] = `
"var foo = function foo() {
  for (var _len = arguments.length, bar = new Array(_len), _key = 0; _key < _len; _key++) {
    bar[_key] = arguments[_key];
  }

  return Promise.resolve(\\"foo \\".concat(bar));
};

module.exports = foo;
"
`;
